tcping.go

```diff
@@ -19,18 +17,72 @@ import (
 	"github.com/google/go-github/v45/github"
 )
 
+// printer is a set of methods for printers to implement.
+//
+// Printers should NOT modify any existing data nor do any calculations.
+// They should only perform visual operations on given data.
+type printer interface {
+	// printStart should print the first message, after the program starts.
+	// This message is printed only once, at the very beginning.
+	printStart(hostname string, port uint16)
+
+	// printProbeSuccess should print a message after each successful probe.
+	// hostname could be empty, meaning it's pinging an address.
+	// streak is the number of successful consecutive probes.
+	printProbeSuccess(hostname, ip string, port uint16, streak uint, rtt float32)
+
+	// printProbeFail should print a message after each failed probe.
+	// hostname could be empty, meaning it's pinging an address.
+	// streak is the number of successful consecutive probes.
+	printProbeFail(hostname, ip string, port uint16, streak uint)
+
+	// printRetryingToResolve should print a message with the hostname
+	// it is trying to resolve an ip for.
+	//
+	// This is only being printed when the -r flag is applied.
+	printRetryingToResolve(hostname string)
+
+	// printTotalDownTime should print a downtime duration.
+	//
+	// This is being called when host was unavailable for some time
+	// but the latest probe was successful (became available).
+	printTotalDownTime(downtime time.Duration)
+
+	// printStatistics should print a message with
+	// helpful statistics information.
+	//
+	// This is being called on exit and when user hits "Enter".
+	printStatistics(s stats)
+
+	// printVersion should print the current version.
+	printVersion()
+
+	// printInfo should a message, which is not directly related
+	// to the pinging and serves as a helpful information.
+	//
+	// Example of such: new version with -u flag.
+	printInfo(format string, args ...any)
+
+	// printError should print an error message.
+	// Printer should also apply \n to the given string, if needed.
+	printError(format string, args ...any)
+}
+
+// currentPrinter is used for outputting information.
+var currentPrinter printer
```

Comment:
I'm not the biggets fan of having this as a global variable. I'd suggest moving it to `processUserInput` (preferably) or the main function.

Reply:
So am I, I even mentioned it in the initial description.

> The `currentPrinter` is now a global variable for `tcpStats` to use. It may be affected by `-j` and similar flags.
> > I'm not sure about this one, yet. ...

Currently `tcpStats` doesn't know anything about the printer, it simply uses it as a global, because the inner field/struct for printer was removed.

But probably the correct way to do this is how it was before, `tcpStats` having a printer inside it.

_I don't really remember the reason why I decided to go for a global, maybe because the JSON printer at the moment was using globals (which is no longer the case as well)_

Reply:
yea, previously it was a bit OOP`ish which I did not like. I am leaning more towards your solution. Anyway I think it can be initialized and set inside the `processUserInput` function. Although I might be wrong.

Reply:
I'm probably missing something here. `currentPrinter` is set to an empty interface here, and actual initialization already happens inside `processUserInput` (last lines)

In the state of `var currentPrinter printer` it's literally not used, it's here so that methods of `tcpStats` could see and call functions from it.

I'm fine either way, just a bit confused which one do you want 😄

Reply:
Ok, I had a closer look and ran some more tests. The current approach, defining the currentPrinter as a global variable makes things a bit easier but I am not the biggest fan of global variables 😄 

I would put it in the `stats` struct. Though needs some changes for some functions in the `tcping.go` file.

Reply:
Sure, will do this way than